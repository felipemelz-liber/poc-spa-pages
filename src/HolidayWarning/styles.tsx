import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  padding: 16px 8px 0px 8px;
  margin-right: auto;
  margin-left: auto;
  @media (min-width: 576px) {
    max-width: 540px;
  }
  @media (min-width: 768px) {
    max-width: 720px;
  }
  @media (min-width: 992px) {
    max-width: 960px;
  }
  @media (min-width: 1200px) {
    max-width: 1140px;
  }
`;

export const FlashAlert = styled.div`
  min-height: 64px;
  border-radius: 4px;
  box-shadow: 0 2px 6px 0 rgba(35, 39, 47, 0.64);
  background-color: #f9fafa;
  margin: 0 4px 24px 4px;
  padding: 12px 12px 12px 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: rgba(35, 39, 47, 0.64);
  border-left: #009dff solid 4px;
`;
