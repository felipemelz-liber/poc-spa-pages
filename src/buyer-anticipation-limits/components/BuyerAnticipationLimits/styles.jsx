import styled, { css } from 'styled-components';
import { handleThemeFromObject } from 'liber-components';
import { TEXT_COLOR } from '../CommonStyles';

const WHITE = '#f9fafa';
const HEADER_COLOR = '#4b6f85';

const getPrimaryColor = ({ theme }) => handleThemeFromObject(theme, 'colors.primary', '#009dff');

export const Container = styled.div`
  width: 100%;

  * {
    font-family: Roboto;
  }
`;

export const Subtitle = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.06px;
  color: ${TEXT_COLOR};
  margin-bottom: 52px;
`;

export const CardsRow = styled.div`
  margin-bottom: 32px;
  display: flex;

  & > div {
    margin-right: 26px;
  }
`;

export const SelectableCard = styled.div`
  width: 200px;
  border-radius: 4px;
  padding: 14px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  svg {
    width: 18px;
    height: 18px;
    margin: 0 8px 4px 0;
  }

  ${({ selected }) =>
    selected
      ? css`
          border: solid 1px ${getPrimaryColor};
          background-color: ${getPrimaryColor};
          * {
            color: ${WHITE} !important;
          }
          svg {
            fill: ${WHITE};
          }
        `
      : css`
          cursor: pointer;
          border: solid 1px #cdd7df;
          svg {
            fill: ${HEADER_COLOR};
          }
        `}
`;

export const Text = styled.div`
  color: ${HEADER_COLOR};
  font-weight: 500;
  font-size: 16px;
`;
