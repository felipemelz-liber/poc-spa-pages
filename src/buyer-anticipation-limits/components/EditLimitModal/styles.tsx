import styled from 'styled-components';
import { FlatButton, TextField } from 'liber-components';

import { TEXT_COLOR } from '../CommonStyles';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const InputText = styled(TextField).attrs(() => ({
  version: 2,
  openLabel: true,
}))`
  min-width: 432px;
  margin-bottom: 32px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;

  & > *:not(:last-child) {
    margin-right: 4px;
  }
  & > *:last-child {
    margin-right: -12px;
  }
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
