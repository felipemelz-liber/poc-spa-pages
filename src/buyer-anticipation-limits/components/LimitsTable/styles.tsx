import styled from 'styled-components';
import { TEXT_COLOR } from '../CommonStyles';

export const ScrollContainer = styled.div`
  width: 100%;
  overflow-x: auto;
  overflow-y: hidden;
  padding-bottom: 160px;
  margin: 24px 0 -80px 0;
`;

export const LimiterContainer = styled.div`
  width: 100%;
  min-width: 930px;
`;

export const ConfirmationContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.06px;
  color: ${TEXT_COLOR};
  width: 408px;
`;
