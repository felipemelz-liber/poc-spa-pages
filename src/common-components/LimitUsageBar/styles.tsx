import styled from 'styled-components';
import { FlatButton } from 'liber-components';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`;

export const Info = styled.div`
  width: 100%;
  padding: 4px 0;
  font-size: 12px;
  line-height: 18px;
  letter-spacing: 0.05px;
  color: #2d2d2d;
`;

export const FullBar = styled.div`
  width: 100%;
  height: 4px;
  border-radius: 4px;
  background-color: #008000;
`;

export const UsedBar = styled(FullBar)`
  background-color: #da2d16;
  width: ${({ usagePercent }) => (usagePercent ? `${usagePercent * 100}%` : 0)};
`;

export const Button = styled(FlatButton)`
  width: fit-content;
  font-size: 12px;
  margin-top: 4px;
  padding: 0 8px 0 4px;

  svg {
    width: 14px;
    height: 14px;
  }
`;
