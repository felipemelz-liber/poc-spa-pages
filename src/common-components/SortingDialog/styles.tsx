import styled from 'styled-components';

export const SectionText = styled.span`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.01px;
  text-align: left;
  color: rgba(35, 39, 47, 0.87);
  margin-bottom: 16px;
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Line = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  margin-bottom: 16px;
`;

export const Wrap = styled(Line)`
  min-width: 332px;
  margin-bottom: 32px;
  flex-wrap: wrap;
`;
