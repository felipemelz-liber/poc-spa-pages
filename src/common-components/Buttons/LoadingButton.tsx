import React, { useRef, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { SuccessButton, LoadingSpinner } from 'liber-components';
import { LoadingContainer } from './LoadingButton.styles';

const LoadingButton = ({
  ButtonComponent,
  children,
  loading,
  onClick,
  disabled,
  size,
  spinnerColor,
  ...rest
}) => {
  const buttonRef = useRef(null);
  const [buttonWidth, setButtonWidth] = useState(0);

  useEffect(() => {
    setButtonWidth(buttonRef.current ? buttonRef.current.offsetWidth : 0);
  }, [buttonRef.current]);

  const getSpinnerSize = () => (size === 'small' ? '14px' : '22px');
  return (
    <ButtonComponent
      ref={buttonRef}
      size={size}
      version={2}
      onClick={onClick}
      disabled={disabled || loading}
      {...rest}
    >
      {loading ? (
        <LoadingContainer minWidth={buttonWidth - 32}>
          <LoadingSpinner color={spinnerColor} size={getSpinnerSize()} />
        </LoadingContainer>
      ) : (
        children
      )}
    </ButtonComponent>
  );
};

LoadingButton.propTypes = {
  ButtonComponent: PropTypes.elementType,
  children: PropTypes.node,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  spinnerColor: PropTypes.string,
};

LoadingButton.defaultProps = {
  ButtonComponent: SuccessButton,
  children: 'BotÃ£o',
  loading: false,
  onClick: () => null,
  disabled: false,
  size: 'medium',
  spinnerColor: 'white',
};

export default LoadingButton;
