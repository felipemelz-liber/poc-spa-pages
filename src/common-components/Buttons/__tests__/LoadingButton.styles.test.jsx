import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { LoadingContainer } from '../LoadingButton.styles';

describe('LoadingButton style tests', () => {
  configure({ adapter: new Adapter() });

  const widths = [0, 10, 55];

  it(`should render LoadingButton without props correctly`, () => {
    const wrapper = shallow(<LoadingContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  widths.forEach(width => {
    it(`should render LoadingButton with width ${width}px correctly`, () => {
      const wrapper = shallow(<LoadingContainer minWidth={width} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
