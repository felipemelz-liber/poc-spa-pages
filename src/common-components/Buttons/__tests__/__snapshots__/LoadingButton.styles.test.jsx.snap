// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton style tests should render LoadingButton with width 0px correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  minWidth={0}
/>
`;

exports[`LoadingButton style tests should render LoadingButton with width 10px correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  minWidth={10}
/>
`;

exports[`LoadingButton style tests should render LoadingButton with width 55px correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  minWidth={55}
/>
`;

exports[`LoadingButton style tests should render LoadingButton without props correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
